### DISCUSSION TITLE
Changing the inverse pole figure default directions

### DISCUSSION THREAD

**Discussion by Erfaaannn** [mtex-toolbox/mtex#1894] [Ask Anything]
Hello everyone
I'm trying to plot an inverse pole figure with my preferred corner directions (such as: [001], [101], [111]). But the default directions are [001], [010], [-1 1 0] for me and I don't know how can I change them. Here is the matlab code I used:

% Load the quaternion data from the text file named 'inc100_HW.txt'
quat = dlmread('inc100_HW.txt');

CS = crystalSymmetry('422');

ori = loadOrientation_generic('inc100_HW.txt','CS',CS, 'ColumnNames', ...
		      {'Quat real','Quat i','Quat j','Quat k'},'Columns',[1,2,3,4],'Bunge');

odf = calcDensity(ori,'resolution',2.5*degree);

plot(odf,'phi2',[0 45 65]* degree,'contour','antipodal')

saveas(gcf,'odf.png'); 

plotPDF(odf,[Miller(1,1,1,CS)],'antipodal','complete','upper')
saveas(gcf,'111_pf.png');

plotIPDF(odf,[xvector],'antipodal')

I will be so thankful if anybody can help me with my problem.

**Comment by zmichels** [comment]
I think you can set the default in the mtex_settings.m file. Specify the a- and b-a is directions relative to the page.On Jul 8, 2023, at 7:27 AM, Erfaaannn ***@***.***> wrote:﻿
Hello everyone
I'm trying to plot an inverse pole figure with my preferred corner directions (such as: [001], [101], [111]). But the default directions are [001], [010], [-1 1 0] for me and I don't know how can I change them. Here is the matlab code I used:
% Load the quaternion data from the text file named 'inc100_HW.txt'
quat = dlmread('inc100_HW.txt');
CS = crystalSymmetry('422');
ori = loadOrientation_generic('inc100_HW.txt','CS',CS, 'ColumnNames', ...
{'Quat real','Quat i','Quat j','Quat k'},'Columns',[1,2,3,4],'Bunge');
odf = calcDensity(ori,'resolution',2.5*degree);
plot(odf,'phi2',[0 45 65]* degree,'contour','antipodal')
saveas(gcf,'odf.png');
plotPDF(odf,[Miller(1,1,1,CS)],'antipodal','complete','upper')
saveas(gcf,'111_pf.png');
plotIPDF(odf,[xvector],'antipodal')
I will be so thankful if anybody can help me with my problem.

**Reply by Erfaaannn** [reply]
Thank you so much for your help. Now I can change the directions of the three corners. But, do you know how can I draw an inverse pole figure with [001], [101], and [111] being the 3 corners of my final result? I tried every possibilities for different specifications of a- and b-a directions in the lines of setMTEXpref('bAxisDirection','east') and setMTEXpref('aAxisDirection',''). As I mentioned, now I can change the default directions but my problem is that I didn't succeed in plotting my inverse pole figure with the directions I mentioned in this message.

**Reply by liuqing72** [reply]
Hi, I have same issues, do you solve it now?

**Comment by kilir** [comment]
Hi,
see here https://github.com/mtex-toolbox/mtex/discussions/1329#discussioncomment-2096427 for example.
Cheers,
Rüdiger
